Lib/test/test_httpservers.py

```diff
@@ -444,6 +444,45 @@ def test_head(self):
         self.assertEqual(response.getheader('content-type'),
                          'application/octet-stream')
 
+    def test_browser_cache(self):
+        #constructs the path relative to the root directory of the HTTPServer
+        response = self.request(self.base_url + '/test')
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+        last_modif = response.headers['Last-modified']
+        
+        # send new request to the same url with request header 
+        # If-Modified-Since set to Last-Modified : must return 304
+        from email.message import Message
+        headers = Message()
+        headers['If-Modified-Since'] = last_modif
+        response = self.request(self.base_url + '/test', headers=headers)
+        self.check_status_and_reason(response, HTTPStatus.NOT_MODIFIED)
+
+        # if If-None-Match header is present, ignore If-Modified-Since
+        headers['If-None-Match'] = "*"
+        response = self.request(self.base_url + '/test', headers=headers)
+        self.check_status_and_reason(response, HTTPStatus.OK)        
+
+        # with If-Modified-Since earlier than Last-Modified, must return 200
+        import datetime
```

Comment:
Import at module level. Importing inside functions is used in exceptional cases.